src/base_alloc/base_alloc_windows.c

```diff
@@ -7,15 +7,24 @@
 
 #include <windows.h>
 
+#include "utils_concurrency.h"
+
+static UTIL_ONCE_FLAG Page_size_is_initialized = UTIL_ONCE_FLAG_INIT;
+static size_t Page_size;
+
 void *ba_os_alloc(size_t size) {
     return VirtualAlloc(NULL, size, MEM_RESERVE | MEM_COMMIT, PAGE_READWRITE);
 }
 
 void ba_os_free(void *ptr, size_t size) { VirtualFree(ptr, 0, MEM_RELEASE); }
 
-size_t ba_os_get_page_size(void) { return 4096; /* TODO */ }
+static void _ba_os_init_page_size(void) {
+    SYSTEM_INFO SystemInfo;
+    GetSystemInfo(&SystemInfo);
+    Page_size = SystemInfo.dwPageSize;
+}
 
-//
-// TODO:
-// 1) implement ba_os_get_page_size()
-//
+size_t ba_os_get_page_size(void) {
```

Comment:
Please rebase (these are not your changes, I believe 😉 )

Reply:
Done